appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleEvent.kt

```diff
@@ -95,4 +107,42 @@ internal sealed class ExperienceLifecycleEvent(
             is ExperienceError -> experienceError
             else -> null
         }
+
+    private val customProperties: HashMap<String, Any>?
+        get() = when (this) {
+            is StepInteraction -> {
+                flatStepIndex?.let {
+                    val step = experience.flatSteps[it]
+                    hashMapOf(
+                        "interactionType" to interactionType.analyticsName(),
+                        "interactionData" to hashMapOf(
+                            "formResponse" to step.formState.formItems.map { formItem ->
+                                formItem.analyticsProperties()
+                            }
+                        ),
+                    )
+                }
+            }
+            else -> null
+        }
+
+    private fun StepInteraction.InteractionType.analyticsName() =
+        when (this) {
+            FORM_SUBMITTED -> "Form Submitted"
+        }
+
+    private fun ExperienceStepFormItemState.analyticsProperties() =
+        hashMapOf<String, Any>(
+            "fieldId" to id.toString(),
+            "fieldType" to type,
+            "fieldRequired" to isRequired,
+            "label" to label,
+            "value" to formattedValues()
+        )
+
+    private fun ExperienceStepFormItemState.formattedValues() =
+        when (this) {
+            is TextInputFormItemState -> value
+            is OptionSelectFormItemState -> values.joinToString(",") // need actual CSV-ifying
```

Comment:
CSV format is a remaining TODO - want to confirm this is OK with platform still

Reply:
@andretortolano thoughts on using this CSV lib to encode? https://github.com/doyaaaaaken/kotlin-csv

We could also roll our own, but need to make sure we have proper handling of values with commas in them, escape double quotes and newlines.

Reply:
ok, will go ahead and progress this one through then cycle back to this.